### PR TITLE
DOC: Clarify Timestamp(unit=) will not necessarily set unit

### LINKED ISSUE #52286: BUG: `pd.TimeStamp` & `pd.Timedelta` not always adhering to `unit` parameter in `pandas-2.0`
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
>>> x = 340.3333333333333
>>> pd.Timestamp(x, unit='us')
Timestamp('1970-01-01 00:00:00.000340333')
>>> pd.Timestamp(x, unit='us').unit
'ns'
>>> pd.Timestamp(x, unit='us').as_unit('us')
Timestamp('1970-01-01 00:00:00.000340')
>>> pd.Timestamp(x, unit='us').as_unit('us').unit
'us'
```

### Issue Description

When creating a `pd.Timestamp` & `pd.Timedelta` with a specific `unit`, the resulting `Timestamp` seems to be always of unit `ns`. We are having to perform `as_unit('us')` to retrieve the desired result.

### Expected Behavior

```python
>>> pd.Timestamp(x, unit='us')
Timestamp('1970-01-01 00:00:00.000340')
```

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : c2a7f1ae753737e589617ebaaff673070036d653
python           : 3.10.10.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.15.0-76-generic
Version          : #86-Ubuntu SMP Fri Jan 17 17:24:28 UTC 2020
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 2.0.0rc1
numpy            : 1.23.5
pytz             : 2023.2
dateutil         : 2.8.2
setuptools       : 67.6.0
pip              : 23.0.1
Cython           : 0.29.33
pytest           : 7.2.2
hypothesis       : 6.70.1
sphinx           : 5.3.0
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.1.2
IPython          : 8.11.0
pandas_datareader: None
bs4              : 4.12.0
bottleneck       : None
brotli           : 
fastparquet      : None
fsspec           : 2023.3.0
gcsfs            : None
matplotlib       : None
numba            : 0.56.4
numexpr          : None
odfpy            : None
openpyxl         : 3.1.2
pandas_gbq       : None
pyarrow          : 11.0.0
pyreadstat       : None
pyxlsb           : None
s3fs             : 2023.3.0
scipy            : 1.10.1
snappy           : 
sqlalchemy       : 1.4.46
tables           : None
tabulate         : 0.9.0
xarray           : None
xlrd             : None
zstandard        : None
tzdata           : None
qtpy             : None
pyqt5            : None

</details>

### PR DESCRIPTION
**Pull request by mroeschke**
- [ ] closes #52286 (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/_libs/tslibs/timestamps.pyx [hunk 1]
```diff
@@ -1293,6 +1293,9 @@ class Timestamp(_Timestamp):
         Unit used for conversion if ts_input is of type int or float. The
         valid values are 'D', 'h', 'm', 's', 'ms', 'us', and 'ns'. For
         example, 's' means seconds and 'ms' means milliseconds.
+
+        The result will be stored in nanoseconds, and the unit attribute
+        will be set as ``'ns'``.
```
Comment (jbrockmendel) [inline pandas/_libs/tslibs/timestamps.pyx:1298]: this is only the case for floats.  for ints we do keep the unit where possible

Comment (lukemanley) [reply pandas/_libs/tslibs/timestamps.pyx:1298]: I noticed this as well (haven't checked to see if its expected):

```
import pandas as pd

unit = "s"

ts1 = pd.to_datetime(1, unit=unit)
ts2 = pd.Timestamp(1, unit=unit)

print(f"{ts1.unit=}, {ts2.unit=}")

td1 = pd.to_timedelta(1, unit=unit)
td2 = pd.Timedelta(1, unit=unit)

print(f"{td1.unit=}, {td2.unit=}")
```

output: 
```
ts1.unit='ns', ts2.unit='s'
td1.unit='ns', td2.unit='ns'
```

Comment (mroeschke) [reply pandas/_libs/tslibs/timestamps.pyx]: Thanks, clarified that it's only for floats

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @mroeschke

**Pull request merged**